lib/matplotlib/image.py

```diff
@@ -909,6 +909,14 @@ def make_image(self, renderer, magnification=1.0, unsampled=False):
         transformed_bbox = TransformedBbox(bbox, trans)
         clip = ((self.get_clip_box() or self.axes.bbox) if self.get_clip_on()
                 else self.get_figure(root=True).bbox)
+        if (
+            isinstance(self._A, np.ndarray)
+            and self._A.ndim == 3
+            and self._A.shape[2] == 3
+            and self._A.dtype == np.uint8
+        ):
+            alpha = np.full(self._A.shape[:2] + (1,), 255, dtype=self._A.dtype)
+            self._A = np.concatenate((self._A, alpha), axis=2)
```

Comment:
There is `_rgb_to_rgba` already in this module, which will also handle the float case.
https://github.com/matplotlib/matplotlib/blob/7bf47316cf618a7e6149856d640c4a4ec56265f2/lib/matplotlib/image.py#L219-L230

I'm also wondering if this is quite the right place for the fix.  Is the problem specific to `AxesImage` or could it affect other subclasses of `_ImageBase`?